src/dvclive/monitor_system.py

```diff
@@ -0,0 +1,192 @@
+                last_metrics = {}
+                try:
+                    last_metrics = self._get_metrics()
+                except psutil.Error:
+                    if self._warn_user:
+                        logger.exception("Failed to monitor CPU metrics")
+                        self._warn_user = False
+
+                self._metrics = merge_with(sum, self._metrics, last_metrics)
+                self._shutdown_event.wait(self._interval)
+                if self._shutdown_event.is_set():
+                    break
+            for name, values in self._metrics.items():
+                blacklisted = any(
+                    name.startswith(prefix) for prefix in self._plot_blacklist_prefix
+                )
+                self._live.log_metric(
+                    name,
+                    values / self._nb_samples,
+                    timestamp=True,
+                    plot=None if blacklisted else self._plot,
+                )
+
+    @abc.abstractmethod
+    def _get_metrics(self) -> Dict[str, Union[float, int]]:
+        pass
+
+    def end(self):
+        self._shutdown_event.set()
+
+
+class CPUMonitor(_SystemMonitor):
+    _plot_blacklist_prefix: Tuple = (
+        METRIC_CPU_COUNT,
+        METRIC_RAM_TOTAL_GB,
+        "system/disk/total (GB)",
+    )
+
+    def __init__(
+        self,
+        interval: float = 0.1,
+        num_samples: int = 20,
+        folders_to_monitor: Optional[Dict[str, str]] = None,
+        plot: bool = True,
+    ):
+        """Monitor CPU resources and log them to DVC Live.
+
+        Args:
+            interval (float): interval in seconds between two measurements.
+                Defaults to 0.5.
+            num_samples (int): number of samples to average. Defaults to 10.
+            folders_to_monitor (Optional[Dict[str, str]]): monitor disk usage
+                statistics about the partition which contains the given paths. The
+                statistics include total and used space in gygabytes and percent.
+                This argument expect a dict where the key is the name that will be used
+                in the metric's name and the value is the path to the folder to monitor.
+                Defaults to {"main": "/"}.
+            plot (bool): should the system metrics be saved as plots. Defaults to True.
+
+        Raises:
+            ValueError: if the arguments passed to the function don't have a
+                supported type.
+        """
+        super().__init__(interval=interval, num_samples=num_samples, plot=plot)
+        folders_to_monitor = (
+            {"main": "/"} if folders_to_monitor is None else folders_to_monitor
+        )
+        self._disks_to_monitor = {}
+        for disk_name, disk_path in folders_to_monitor.items():
+            if disk_name != os.path.normpath(disk_name):
+                raise ValueError(  # noqa: TRY003
+                    "Keys for `partitions_to_monitor` should be a valid name"
```

Comment:
hmm, what is the valid name in this case?

Reply:
I would still move try catch to where we use ... e.g. someone drops the directory, it should not be causing other metrics to disappear